pkg/pillar/scripts/device-steps.sh

```diff
@@ -395,7 +445,9 @@ if [ $SELF_REGISTER = 1 ]; then
         exit 1
     fi
     echo "$(date -Ins -u) Starting client selfRegister getUuid"
-    if ! $BINDIR/client selfRegister getUuid; then
+    $BINDIR/client selfRegister getUuid
+    wait_for_file $UUID_FILE 2147483647 #wait forever(68 years) for the uuid file. 2147483647 is max int value in 32 bit OS.
```

Comment:
Why can't you wait for client.done here?

Reply:
I assumed that was one of the forever running agent.
Changed.